mailpoet/tests/DataFactories/NewsletterOption.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace MailPoet\Test\DataFactories;
+
+use MailPoet\DI\ContainerWrapper;
+use MailPoet\Entities\NewsletterEntity;
+use MailPoet\Entities\NewsletterOptionEntity;
+use MailPoet\Entities\NewsletterOptionFieldEntity;
+use MailPoet\Newsletter\Options\NewsletterOptionFieldsRepository;
+use MailPoet\Newsletter\Options\NewsletterOptionsRepository;
+use MailPoetVendor\Doctrine\ORM\EntityManager;
+
+class NewsletterOption {
+  public function createMultipleOptions(NewsletterEntity $newsletter, array $options) {
+    foreach ($options as $optionField => $optionValue) {
+      $this->create($newsletter, $optionField, $optionValue);
+    }
+  }
+
+  public function create(NewsletterEntity $newsletter, string $optionField, string $optionValue): NewsletterOptionEntity {
+    $entityManager = ContainerWrapper::getInstance()->get(EntityManager::class);
+    $newsletterOptionFieldRepository = ContainerWrapper::getInstance()->get(NewsletterOptionFieldsRepository::class);
+    $newsletterOptionRepository = ContainerWrapper::getInstance()->get(NewsletterOptionsRepository::class);
+
+    $newsletterOptionField = $newsletterOptionFieldRepository->findOneBy(['name' => $optionField]);
+    if (!$newsletterOptionField instanceof NewsletterOptionFieldEntity) {
+      $newsletterOptionField = new NewsletterOptionFieldEntity();
+      $newsletterOptionField->setName($optionField);
+      $newsletterOptionField->setNewsletterType($newsletter->getType());
+      $entityManager->persist($newsletterOptionField);
```

Comment:
That is good to know. Thanks for mentioning it, @costasovo. In this particular case, I think I will leave the code as is since I'm dealing with two different repositories. I know that calling flush() on one of the repositories would be enough to flush the changes from both repositories, but it seems counter-intuitive to me, and thus I prefer to keep using `$entityManager->flush()` (unless there is some disadvantage to using `$entityManager` in this context that I'm missing).